libs/client/src/lib.rs

```diff
@@ -447,8 +447,14 @@ impl KanidmClient {
         &self.origin
     }
 
+    /// Returns the base URL of the server
     pub fn get_url(&self) -> &str {
-        self.addr.as_str()
+        &self.addr
```

Comment:
I wonder if it's worth making this a URL type since then we can do .join for safe append of endpoints.

Reply:
Guess it depends if reqwest takes a url or str? If url supports .as_str() then it's probably worth.